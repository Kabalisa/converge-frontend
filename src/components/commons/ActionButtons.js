import React from 'react';
import PropTypes from 'prop-types';
import Spinner from '../commons/Spinner';

import '../../assets/styles/actionbuttons.scss';

/**
 * Reusable button component for performing
 * actions and canceling
 *
 * @param {array} props
 *
 * @returns {JSX}
 */
const ActionButtons = props => (
  props.isLoading ? <Spinner size="small" /> :
  <div className="form-action-buttons">
    <button className="cancel-button action-button" onClick={props.onClickCancel}>
      {props.cancelButtonText}
    </button>
    <button className="action-button" onClick={props.onClickSubmit}>
      {props.actionButtonText}
    </button>
  </div>

);

ActionButtons.propTypes = {
  onClickCancel: PropTypes.func.isRequired,
  cancelButtonText: PropTypes.string,
  actionButtonText: PropTypes.string,
  isLoading: PropTypes.bool,
  onClickSubmit: PropTypes.func,
};

ActionButtons.defaultProps = {
  cancelButtonText: 'CANCEL',
  actionButtonText: 'SAVE CHANGES',
  isLoading: false,
  onClickSubmit: PropTypes.func,
};

export default ActionButtons;
