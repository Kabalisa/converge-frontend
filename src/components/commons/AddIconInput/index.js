import React, { Component, Fragment } from 'react';
import uuid from 'uuid';
import PropTypes from 'prop-types';
import SingleInputWithAddIcon from './InputWithPlusIcon';

class InputsWithAddIcons extends Component {
  constructor(props) {
    super(props);
    this.textInputs = [
      <SingleInputWithAddIcon
        pressed={() => {
          this.onAddInput();
        }}
        onInputChange={props.onInputChange}
        id={uuid()}
      />,
    ];
    this.state = {
      displayInputs: this.textInputs,
    };
  }

  onAddInput = () => {
    this.textInputs.push(
      <SingleInputWithAddIcon
        pressed={() => {
          this.onAddInput();
        }}
        onInputChange={this.props.onInputChange}
        id={uuid()}
      />,
    );
    this.setState({
      displayInputs: this.textInputs,
    });
  };

  render() {
    const { displayInputs } = this.state;
    return <Fragment>{displayInputs}</Fragment>;
  }
}

InputsWithAddIcons.propTypes = {
  onInputChange: PropTypes.func.isRequired,
};
export default InputsWithAddIcons;
