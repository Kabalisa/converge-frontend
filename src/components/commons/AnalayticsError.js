import React from 'react';
import PropTypes from 'prop-types';
import ErrorIcon from '../commons/ErrorIcon';

const AnalyticsError = ({ message }) => (
  <article className="pie-chart">
    <ErrorIcon message={message} />
  </article>
);

AnalyticsError.propTypes = {
  message: PropTypes.string,
};

AnalyticsError.defaultProps = {
  message: '',
};
export default AnalyticsError;
