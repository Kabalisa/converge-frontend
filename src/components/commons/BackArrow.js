import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import '../../assets/styles/backArrow.scss';

const BackArrow = ({ redirectUri }) => (
  <NavLink to={redirectUri} className="arrow">
    <span className="arrow-symbol">
      &#11013;
    </span>
    <span>Back</span>
  </NavLink>
);

BackArrow.propTypes = {
  redirectUri: PropTypes.string.isRequired,
};

export default BackArrow;
