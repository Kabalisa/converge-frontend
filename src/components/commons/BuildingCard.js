/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';

import '../../assets/styles/officeStructure.scss';

const setClassName = (block, activeBlock) => {
  const className = `office-structure-block ${
    block === activeBlock
      ? 'office-structure-block active-block'
      : 'office-structure-block'
  }`;
  return className;
};
const BuildingCard = ({ blocks, activeBlock, toggleBlock }) => (
  <div className="office-structure-container">
    {blocks.map(block => (
      <button
        className={setClassName(block.name, activeBlock)}
        onClick={() => toggleBlock(block.name)}
      >
        {block.name}
      </button>
    ))}
  </div>
);

BuildingCard.propTypes = {
  blocks: PropTypes.array,
  activeBlock: PropTypes.string,
  toggleBlock: PropTypes.func.isRequired,
};

BuildingCard.defaultProps = {
  blocks: [],
  activeBlock: '',
};

export default BuildingCard;
