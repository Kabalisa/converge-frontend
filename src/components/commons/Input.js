import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/input.scss';
import Controls from '../helpers/Controls';

export class Input extends Component {
  constructor(props) {
    super(props);
    this.inputRef = React.createRef();
  }

  clear = () => {
    this.inputRef.current.value = 0;
  };

  /**
   * Increases and decreases numeric input value
   *
   * @param {object} event
   *
   * @returns {void}
   */
  handleIncrement = (event) => {
    const { onChange } = this.props;
    event.preventDefault();

    /* Logic for increasing and decreasing numeric input value
     * should be placed here.
     */

    const num = parseInt(this.inputRef.current.value, 10);
    if (event.target.name === 'up') {
      /* The button up event and Incremented number is passed
       * to the onChange function when the up button is
       * pressed.
       */
      onChange(event, num + 1);
      this.inputRef.current.value = num + 1;
    }
    if (event.target.name === 'down') {
      if (num >= 1) {
        onChange(event, num - 1);
        this.inputRef.current.value = num - 1;
      }
    }
  };

  render() {
    const {
      name,
      value,
      inputClass,
      type,
      labelName,
      labelClass,
      id,
      placeholder,
      onChange,
      controlsClass,
      ...otherProps
    } = this.props;

    return (
      <div className={labelClass}>
        <label htmlFor={name}>
          {labelName}
          <input
            ref={this.inputRef}
            type={type}
            className={inputClass}
            name={name}
            value={value}
            onChange={onChange}
            placeholder={placeholder}
            id={id}
            {...otherProps}
          />

          {type === 'number' ? (
            <Controls
              controlsClass={controlsClass}
              handleIncrement={this.handleIncrement}
            />
          ) : null}
        </label>
      </div>
    );
  }
}

Input.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  inputClass: PropTypes.string,
  controlsClass: PropTypes.string,
  id: PropTypes.string.isRequired,
  labelName: PropTypes.string.isRequired,
  labelClass: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  type: PropTypes.string,
};

Input.defaultProps = {
  value: 'text',
  type: 'text',
  inputClass: 'mrm-input default-input',
  labelClass: 'input1',
  controlsClass: '',
  placeholder: '',
};

export default Input;
