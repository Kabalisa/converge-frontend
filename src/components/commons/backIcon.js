/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';
import backIcon from '../../assets/images/ic_back.svg';

const BackIcon = ({ handleOnClick, page }) => (
  <div className="back_icon" onClick={() => handleOnClick(page)}>
    <img src={backIcon} alt="back icon" />
  </div>
);

BackIcon.propTypes = {
  handleOnClick: PropTypes.func.isRequired,
  page: PropTypes.string.isRequired,
};

export default BackIcon;
