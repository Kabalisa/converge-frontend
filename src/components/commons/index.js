import SelectInput from './SelectInput';
import Modal from './Modal';
import { Input } from './Input';
import ActionButtons from './ActionButtons';

export { SelectInput, Modal, Input, ActionButtons };
