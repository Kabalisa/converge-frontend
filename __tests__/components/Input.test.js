import React from 'react';
import { shallow } from 'enzyme';
import { Input } from '../../src/components/commons/Input';

describe('Input Component', () => {
  const props = {
    name: '',
    value: '',
    inputClass: '',
    id: '',
    labelName: '',
    labelClass: '',
  };
  const wrapper = shallow(<Input {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have one input', () => {
    const modalForm = wrapper.find('input');
    expect(modalForm).toHaveLength(1);
  });

  it('should have one label', () => {
    const modalForm = wrapper.find('label');
    expect(modalForm).toHaveLength(1);
  });
});
