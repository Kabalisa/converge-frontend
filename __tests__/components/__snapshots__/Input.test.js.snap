// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders properly 1`] = `
<div
  className=""
>
  <label
    htmlFor=""
  >
    <input
      className=""
      id=""
      name=""
      onChange={[MockFunction]}
      type="text"
      value=""
    />
  </label>
</div>
`;
